lib/surface/compiler.ex

```diff
@@ -94,7 +95,8 @@ defmodule Surface.Compiler do
       line: line,
       file: file,
       caller: caller,
-      checks: opts[:checks] || []
+      checks: opts[:checks] || [],
+      variables: opts[:variables] || []
```

Comment:
I'm not happy adding this field to `CompileMeta` but I couldn't find another way to pass the list of existing variables to be unquoted by `surface_quote`.

Reply:
I'm going to spend a bit of time messing around with some things to see if I can come up with an alternative, but I'm not positive currently